configapi/api_inventory.go

```diff
@@ -0,0 +1,231 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2024 Canonical Ltd.
+
+package configapi
+
+import (
+    "errors"
+    "fmt"
+    "encoding/json"
+    "strings"
+    "net/http"
+
+    "github.com/gin-gonic/gin"
+    "github.com/omec-project/util/httpwrapper"
+    "github.com/omec-project/webconsole/backend/logger"
+    "github.com/omec-project/webconsole/configmodels"
+    "github.com/omec-project/webconsole/dbadapter"
+    "go.mongodb.org/mongo-driver/bson"
+)
+
+const (
+    gnbDataColl = "webconsoleData.snapshots.gnbData"
+    upfDataColl = "webconsoleData.snapshots.upfData"
+)
+
+func setInventoryCorsHeader(c *gin.Context) {
+    c.Writer.Header().Set("Access-Control-Allow-Origin", "*")
+    c.Writer.Header().Set("Access-Control-Allow-Credentials", "true")
+    c.Writer.Header().Set("Access-Control-Allow-Headers", "Content-Type")
+    c.Writer.Header().Set("Access-Control-Allow-Methods", "POST, GET, DELETE")
+}
+
+func GetGnbs(c *gin.Context) {
+    setInventoryCorsHeader(c)
+    logger.WebUILog.Infoln("Get all gNBs")
+
+    var gnbs []*configmodels.Gnb
+    gnbs = make([]*configmodels.Gnb, 0)
+    rawGnbs, errGetMany := dbadapter.CommonDBClient.RestfulAPIGetMany(gnbDataColl, bson.M{})
+    if errGetMany != nil {
+        logger.DbLog.Errorln(errGetMany)
+        c.JSON(http.StatusInternalServerError, gnbs)
+    }
+
+    for _, rawGnb := range rawGnbs {
+        var gnbData configmodels.Gnb
+        err := json.Unmarshal(mapToByte(rawGnb), &gnbData)
+        if err != nil {
+            logger.DbLog.Errorf("Could not unmarshall gNB %v", rawGnb)
+        }
+        gnbs = append(gnbs, &gnbData)
+    }
+    c.JSON(http.StatusOK, gnbs)
+}
+
+func GnbPostByName(c *gin.Context) {
```

Comment:
I'd encourage standardising with the name of the functions. For instance `GetGnbs` starts with the http method (GET) and ends with the object (Gnbs) vs. in `GnbPostByName` where we start with the Object (Gnb) then end with the action (Get).

I'd recommend the following names: `GetGnbs`, `PostGnb` and `DeleteGnb`